src/rpc/mining.cpp

```diff
@@ -465,7 +465,8 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
     if(!g_connman)
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
 
-    if (g_connman->GetNodeCount(CConnman::CONNECTIONS_ALL) == 0)
+    // Allow getblocktemlate for not connected regtest node
+    if (Params().NetworkIDString() != "regtest" && g_connman->GetNodeCount(CConnman::CONNECTIONS_ALL) == 0)
```

Comment:
Do the regtest comparison second so it can't incur *any* penalty on normal operation.